lib/workers/repository/update/branch/index.ts

```diff
@@ -73,17 +77,38 @@ async function deleteBranchSilently(branchName: string): Promise<void> {
 }
 
 export interface ProcessBranchResult {
+  configAndManagersHash?: string;
   branchExists: boolean;
   prBlockedBy?: PrBlockedBy;
   prNo?: number;
   result: BranchResult;
 }
 
+function canSkipBranchUpdateCheck(
+  config: BranchConfig,
+  branchCache: BranchCache,
+  configAndManagersHash: string
+): boolean {
+  const currentBaseBranchSha = getBranchCommit(config.baseBranch!);
+  const branchCommit = getBranchCommit(config?.branchName);
+
+  return (
+    currentBaseBranchSha === branchCache?.parentSha &&
```

Comment:
Do we really need to check this? If the parent SHA had changed, and that was important, wouldn't we have rebased already?

Reply:
If the rebase always happens inside the processBranch function then this check is necessary, because the code logic which handles rebasing comes after this check. So if we remove this check then we won't be able to detect that a branch needs rebasing.

Reply:
Okay, so the position for this check needs to be changed in this case. Any other conditions that we should be doing before checking the cache?

Reply:
The reason for the comparing branch & parent SHAs was to check if we have any of these 3 conditions:
1. conflicted branch
2. behindBaseBranch
3. modified branch
We need to clone and do the needful if any of the condition is true thus can't reuse the cache. But since we now use cache internally for all these checks these both checks aren't necessary.

Reply:
What's a scenario where they could be different? e.g. how would it be that the branch SHA is identical but there's a mismatch of parent sha?

Reply:
If branch is behind base then parentSha will differ but branchSHA won't.

Reply:
You're right, I got temporarily confused. But to skip the updates check we don't need for the parent SHA to equal current base branch SHA. I think this check should come after rebaseWhen checks and not care about parent SHA.